hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Schedulable.java

```diff
@@ -29,5 +30,9 @@
 public interface Schedulable {
   public UserGroupInformation getUserGroupInformation();
 
+  default CallerContext getCallerContext() {
+    throw new NotImplementedException("Code is not implemented");
```

Comment:
i worry about exposing commons.* classes in an API, as if we change what is raised external code may break. Just raise java's UnsupportedOperationException and mention this may happen in the javadocs